go-controller/pkg/util/gateway-init.go

```diff
@@ -46,68 +46,81 @@ func getLocalSystemID() (string, error) {
 	return localSystemID, nil
 }
 
-func lockNBForGateways() error {
-	localSystemID, err := getLocalSystemID()
+// GetDefaultGatewayRouterIP returns the first gateway logical router name
+// and IP address as listed in the OVN database
+func GetDefaultGatewayRouterIP() (string, net.IP, error) {
+	stdout, stderr, err := RunOVNNbctl("--data=bare", "--format=table",
+		"--no-heading", "--columns=name,options", "find", "logical_router",
+		"options:lb_force_snat_ip!=-")
 	if err != nil {
```

Comment:
@dcbw L52-54 is going to return different rows based on the state of the cluster. For example:

I had two node K8s-cluster and the above `ovn-nbctl` command returned the following
```
$ ovn-nbctl --data=bare --format=table --no-heading --columns=name,options find logical_router "options:lb_force_snat_ip!=-"
GR_k8s-node1 chassis=1efaa0cc-68e2-4b10-80a7-f7ee841d162e lb_force_snat_ip=100.64.0.3
GR_k8s-node2 chassis=86d63482-d1fb-45f1-8659-094cd0cac98a lb_force_snat_ip=100.64.0.2
```
So, GR_k8s-node1 will be used for the default gateway router. Now, I added a third node `k8s-node1` and the above output changed to

```
$ ovn-nbctl --data=bare --format=table --no-heading --columns=name,options find logical_router "options:lb_force_snat_ip!=-"
GR_k8s-node3 chassis=ea15c77d-8552-4a6c-9c2b-650a1dc3c885 lb_force_snat_ip=100.64.0.4
GR_k8s-node2 chassis=1efaa0cc-68e2-4b10-80a7-f7ee841d162e lb_force_snat_ip=100.64.0.3
GR_k8s-node1 chassis=86d63482-d1fb-45f1-8659-094cd0cac98a lb_force_snat_ip=100.64.0.2
```

So, going forward k8s-node3 would be the default gateway router....

Reply:
Fixed in latest push.